src/AWSLambda/Events/SNSEvent.hs

```diff
@@ -50,7 +44,19 @@ data SNSMessage message = SNSMessage
   } deriving (Eq, Show, Generic)
 
 instance FromText message => FromJSON (SNSMessage message) where
-  parseJSON = genericParseJSON $ aesonDrop 3 pascalCase
+  parseJSON = withObject "SNSMessage'" $ \o ->
```

Comment:
Might be worth adding a comment to this instance to explain why it is the way it is - without knowing the history it looks a bit weird.

Reply:
done